pkg/tls/config.go

```diff
@@ -3,25 +3,22 @@ package tls
 import (
 	"crypto/tls"
 	"fmt"
+	"sort"
 	"strings"
-
-	"github.com/rancher/rancher/pkg/settings"
-)
-
-const (
-	httpsMode = "https"
-	acmeMode  = "acme"
 )
 
 var (
-	tlsVersions = map[string]uint16{
+	validVersions = map[string]uint16{
+		"1.3": tls.VersionTLS13,
 		"1.2": tls.VersionTLS12,
+		// Deprecated.
 		"1.1": tls.VersionTLS11,
+		// Deprecated.
 		"1.0": tls.VersionTLS10,
 	}
 
 	// https://golang.org/pkg/crypto/tls/#pkg-constants
-	tlsCipherSuites = map[string]uint16{
+	validCiphers10to12 = map[string]uint16{
```

Comment:
Yes but you could set the 1.3 versions apart by having a 1.3 for that variable, and leaving this one alone. See lines 149 and 201 in https://go.dev/src/crypto/tls/cipher_suites.go

Reply:
I think having those two slices with numbers in names is just more explicit. Besides, in the code listing you provided, I see variables like `supportedUpToTLS12` and `defaultCipherSuitesTLS13`, so I don't think our code is out of line with how that code lists different ciphers.

Reply:
I don't mind the name change for this case because being explicit improves readability. Since you don't have to rely on code comments to explain version compatibility